File: Aplicacion mobile/SensoTrack/app/src/main/java/com/example/sensotrack/room/SensorData.kt
import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "sensor_data")
data class SensorData(
    @PrimaryKey(autoGenerate = true) val id: Int = 0,
    val sensorType: String,         // Tipo de sensor (SCREEN, LOCATION, etc.)
    val action: String?,            // Acción, si es aplicable (ejemplo: ACTION_USER_PRESENT para la pantalla)
    val latitude: Double?,          // Coordenadas de ubicación, si aplica
    val longitude: Double?,
    val lightValue: Float?,         // Valor de luz, si aplica
    val timestamp: String           // Marca de tiempo de la recolección del dato
)

File: HealthSense/src/models/SensorData.ts
import { DataTypes } from 'sequelize';
import { sequelize } from './sequelize';

export const SensorData = sequelize.define('SensorData', {
  id: {
    type: DataTypes.INTEGER,
    primaryKey: true,
    autoIncrement: true,
  },
  sensorId: {
    type: DataTypes.INTEGER,
    allowNull: false,
  },
  sensorType: {
    type: DataTypes.STRING,
    allowNull: false,
  },
  timestamp: {
    type: DataTypes.DATE,
    allowNull: false, 
  },
  lightValue: {
    type: DataTypes.INTEGER,
    allowNull: true,
  },
  latitude: {
    type: DataTypes.DOUBLE,
    allowNull: true,
  },
  longitude: {
    type: DataTypes.DOUBLE,
    allowNull: true,
  },
  action: {
    type: DataTypes.STRING,
    allowNull: true,
  },
}, {
  tableName: 'SensorData',
});
